packages/@aws-cdk/aws-cognito/lib/user-pool-attr.ts

```diff
@@ -141,6 +141,66 @@ export interface CustomAttributeConfig {
    * @default - None.
    */
   readonly numberConstraints?: NumberAttributeConstraints;
+
+  /**
+   * Specifies whether the value of the attribute can be changed.
+   * For any user pool attribute that's mapped to an identity provider attribute, you must set this parameter to true.
+   * Amazon Cognito updates mapped attributes when users sign in to your application through an identity provider.
+   * If an attribute is immutable, Amazon Cognito throws an error when it attempts to update the attribute.
+   *
+   * @default false
+   */
+  readonly mutable?: boolean
+
+}
+
+/**
+ * Constraints that can be applied to a custom attribute of string type.
+ */
+export interface CustomAttributeProps {
+
+  /**
+   * Specifies whether the value of the attribute can be changed.
+   * For any user pool attribute that's mapped to an identity provider attribute, you must set this parameter to true.
+   * Amazon Cognito updates mapped attributes when users sign in to your application through an identity provider.
+   * If an attribute is immutable, Amazon Cognito throws an error when it attempts to update the attribute.
+   *
+   * @default false
+   */
+  readonly mutable?: boolean
+}
+
+/**
+ * The base class for all custom attribute types.
+ * All common properties are set here and the method `baseAttributeConfig`
+ * should be used by subclasses to create base CustomAttributeConfig object inside the `bind()` method.
+ */
+export abstract class CustomAttribute implements ICustomAttribute {
```

Comment:
I don't see much value in this base class anymore, now that we've dropped the adminOnly property. 

It would be simpler to just store the `mutable` value as an instance property in each class and return it as part of `bind()`. And to revert the data type to how it was modeled previously.

What do you think?